lighthouse-core/audits/seo/anchor-text.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit');
+const URL = require('../../lib/url-shim');
+const BLOCKLIST = new Set([
+  'click here',
+  'click this',
+  'go',
+  'here',
+  'this',
+  'start',
+  'right here',
+  'more',
+  'learn more',
+]);
+
+class AnchorText extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Content Best Practices',
+      name: 'anchor-text',
+      description: 'Anchors have descriptive text.',
+      failureDescription: 'Anchors do not have descriptive text',
+      helpText: 'Descriptive anchor text helps search engines understand your content. ' +
+      '[Learn more](https://webmasters.googleblog.com/2008/10/importance-of-link-architecture.html)',
+      requiredArtifacts: ['URL', 'CrawlableAnchors'],
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    const failingAnchors = artifacts.CrawlableAnchors
+      .filter(anchor => {
+        if (
+          anchor.href.toLowerCase().startsWith('javascript:') ||
+          URL.equalWithExcludedFragments(anchor.href, artifacts.URL.finalUrl)
```

Comment:
why are links within the page not counted?

Reply:
As far as I understand, these links have no SEO value and often are handled by JS anyway (e.g. `<a href='#'>show more</a>` to expand some text inline).